Marlin/Marlin_main.cpp

```diff
@@ -5038,7 +5038,7 @@ inline void gcode_M121() { endstops.enable_globally(false); }
    */
   inline void gcode_M156() {
     uint8_t addr = code_seen('A') ? code_value_short() : 0;
-    int bytes    = code_seen('B') ? code_value_short() : 0;
+    int bytes    = code_seen('B') ? code_value_long() : 0;
 
     if (addr && bytes) {
       i2c.address(addr);
```

Comment:
Tis true… But they don’t want to hog the computer up… So THOSE poor programmers have to do it
packet by packet the old fashioned way…

Man, You are a MANIAC, and GOOD. 

> On May 10, 2016, at 12:15 PM, Scott Lahteine <notifications@github.com> wrote:
> 
> In Marlin/Marlin_main.cpp <https://github.com/MarlinFirmware/Marlin/pull/3713#discussion_r62702720>:
> 
> > @@ -5038,7 +5038,7 @@ inline void gcode_M121() { endstops.enable_globally(false); }
> >     */
> >    inline void gcode_M156() {
> >      uint8_t addr = code_seen('A') ? code_value_short() : 0;
> > -    int bytes    = code_seen('B') ? code_value_short() : 0;
> > +    int bytes    = code_seen('B') ? code_value_long() : 0;
> >  
> >      if (addr && bytes) {
> >        i2c.address(addr);
> Ah, very good. It seemed to me that somewhere in time and space there might be a request for more than 65,535 bytes over an i2c bus, but in hindsight, quite right, that would be a bit much.
> 
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub <https://github.com/MarlinFirmware/Marlin/pull/3713/files/f1ea661b05d5bb6600859a92d8eb295d809ee2c7#r62702720>